src/components/HintModal.jsx

```diff
@@ -0,0 +1,79 @@
+import { Button, Paragraph } from 'grommet';
+import injectSheet from 'react-jss';
+import React, { Component } from 'react';
+import Modal from 'react-modal';
+
+class HintModal extends Component {
+    constructor() {
+        super();
+        this.state = {
+            showModal: false
```

Comment:
can we change this to `shouldShowModal`? I like to keep action variable names for things that are functions.